/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package singa;

// TODO(wangwei) check protobuf version to include the syntax
//syntax = "proto2";

enum DataType {
  kFloat32 = 0;
  kFloat16 = 1;
  kInt = 2;
  kChar = 3;
  kDouble = 4;
  kNumDataType = 5;
}

enum LangType {
  kCpp = 0;
  kCuda = 1;
  kOpencl = 2;
  kNumDeviceType = 4;
}

enum CopyDirection {
  kHostToHost = 0;
  kHostToDevice = 1;
  kDeviceToHost = 2;
  kDeviceToDevice = 3;
  kNumDirection = 4;
}
